pkgs/kswitch/kswitch.sh

```diff
@@ -47,6 +49,29 @@ _continue() {
   [[ $REPLY =~ ^[Yy]$ ]] || exit 1
 }
 
+wait_lock() {
+    local lock
+    lock="${lockPath}/$*"
+    log_debug "Waiting for lock: ${lock}"
+    while ! mkdir "${lock}" 2>/dev/null; do :;done
+    log_debug "Lock acquired: ${lock}"
+}
+
+lock() {
+    local lock
+    lock="${lockPath}/$*"
+    log_debug "Trying to acquire lock: ${lock}"
+    mkdir "${lock}" 2>/dev/null && log_debug "Lock acquired: ${lock}" && return 0
+    log_debug "Lock unavailable: ${lock}" && return 1
+}
+
+unlock() {
+    local lock
+    lock="${lockPath}/$*"
+    rm -r "${lock}" 2>/dev/null && log_debug "Lock released: ${lock}" && return 0
+    log_debug "Cannot release lock: ${lock}" && return 1
+}
```

Comment:
Why not use a binary designed to properly handle lock files like `flock` ?

Reply:
Tried it but didn't really liked having to handle fileDescriptor + some issue to build it at latest version with nix (required for exclusive lock).
Could still be changed if I build it successfully.